iml-manager-cli/src/server.rs

```diff
@@ -218,250 +219,400 @@ async fn wait_till_agent_starts(
     Ok(())
 }
 
-#[allow(clippy::cognitive_complexity)]
-pub async fn server_cli(command: ServerCommand) -> Result<(), ImlManagerCliError> {
-    match command {
-        ServerCommand::List => {
-            let hosts: ApiList<Host> = wrap_fut("Fetching hosts...", get_hosts()).await?;
+fn list_server(hosts: ApiList<Host>) {
+    tracing::debug!("Hosts: {:?}", hosts);
+
+    let table = generate_table(
+        &["Id", "FQDN", "State", "Nids"],
+        hosts.objects.into_iter().map(|h| {
+            vec![
+                h.id.to_string(),
+                h.fqdn,
+                h.state,
+                h.nids.unwrap_or_default().join(" "),
+            ]
+        }),
+    );
+
+    table.printstd();
+}
+
+fn get_profile(config: &AddHosts, profiles: ApiList<ServerProfile>) -> Option<ServerProfile> {
+    let profile_opt = profiles
+        .objects
+        .into_iter()
+        .filter(|x| x.user_selectable)
+        .find(|x| x.name == config.profile);
+
+    if profile_opt.is_none() {
+        display_error(format!("Unknown profile {}", config.profile));
```

Comment:
It might be better to separate pure code from none-pure :) Especially when you handle `None` anyway.

Reply:
The initial problem was the high complexity count (in the mid to upper 50's). In an effort to reduce the complexity I moved this if statement inside the `get_profile` function. I can move the condition to the `profiles_and_servers` function if you prefer.